src/archivist/fetcher/index.js

```diff
@@ -4,10 +4,23 @@ import fetchHtmlOnly from './htmlOnlyFetcher.js';
 export { launchHeadlessBrowser, stopHeadlessBrowser } from './fullDomFetcher.js';
 export { FetchDocumentError } from './errors.js';
 
-export default async function fetch({ url, executeClientScripts, cssSelectors }) {
+/**
+ * Fetch url using headless browser or not and waiting for specific selectors to exist
+ *
+ * @param {Object} config - The fetcher config
+ * @param {string} config.url - The url to be fetched
+ * @param {boolean} [config.executeClientScripts] - Whether you want to use a headless browser are not
+ * @param {string[]} [config.cssSelectors] - List of css selectors to await for (only when executeClientScripts is true)
+ * @param {Object} [config.options] - The fetcher options
+ * @param {number} [config.options.navigationTimeout=5000] - Time to wait before considering the fetch failed
+ * @param {string} [config.options.language=en] - The language to be passed in headers
+ * @param {number} [config.options.waitForElementsTimeout=5000] - Time to wait for selectors to exist on page before considering the fetch failed (only when executeClientScripts is true)
+ * @returns {Promise} Promise object with `mimeType` and `content`
+ */
+export default async function fetch({ url, executeClientScripts, cssSelectors, options }) {
   if (executeClientScripts) {
-    return fetchFullDom(url, cssSelectors);
+    return fetchFullDom(url, cssSelectors, options);
```

Comment:
Define at this level options common to the two fetchers to ensure consistency: `navigationTimeout` and `language`.

Reply:
Not sure what you mean.

Do you want defaults to be defined here?
Or only pass relevant options to fetchHtmlOnly with a `pick` for example?

Thanks for clarifying